Infection `develop` fails

See https://github.com/codeigniter4projects/website/actions/workflows/infection.yml?query=branch%3Adevelop

```
Run git fetch --depth=1 origin $GITHUB_BASE_REF
  git fetch --depth=1 origin $GITHUB_BASE_REF
  infection --threads=2 --skip-initial-tests --coverage=build/phpunit --git-diff-base=origin/$GITHUB_BASE_REF --git-diff-filter=AM --logger-github --ignore-msi-with-no-mutations
  shell: /usr/bin/bash -e {0}
  env:
    COMPOSER_PROCESS_TIMEOUT: 0
    COMPOSER_NO_INTERACTION: 1
    COMPOSER_AUTH: {"github-oauth": {"github.com": "***"}}
From https://github.com/codeigniter4projects/website
 * [new branch]      dependabot/composer/codeigniter4/devkit-1.1.0 -> origin/dependabot/composer/codeigniter4/devkit-1.1.0
 * [new branch]      dependabot/composer/codeigniter4/framework-4.2.6 -> origin/dependabot/composer/codeigniter4/framework-4.2.6
 * [new branch]      master     -> origin/master

In Process.php line 112:
                                                                               
  The command "'git' 'diff' 'origin/' '--diff-filter' 'AM' '--name-only' '--'  
   'app/'" failed.                                                             
                                                                               
  Exit Code: 128(Invalid exit argument)                                        
                                                                               
  Working directory: /home/runner/work/website/website                         
                                                                               
  Output:                                                                      
  ================                                                             
                                                                               
                                                                               
  Error Output:                                                                
  ================                                                             
  fatal: bad revision 'origin/'                                                
                                                                               

run [--test-framework TEST-FRAMEWORK] [--test-framework-options TEST-FRAMEWORK-OPTIONS] [-j|--threads THREADS] [--only-covered] [-s|--show-mutations] [--no-progress] [--force-progress] [-c|--configuration CONFIGURATION] [--coverage COVERAGE] [--mutators MUTATORS] [--filter FILTER] [--formatter FORMATTER] [--git-diff-filter GIT-DIFF-FILTER] [--git-diff-lines] [--git-diff-base GIT-DIFF-BASE] [--logger-github [LOGGER-GITHUB]] [--logger-html LOGGER-HTML] [--noop] [--only-covering-test-cases] [--min-msi MIN-MSI] [--min-covered-msi MIN-COVERED-MSI] [--log-verbosity LOG-VERBOSITY] [--initial-tests-php-options INITIAL-TESTS-PHP-OPTIONS] [--skip-initial-tests] [--ignore-msi-with-no-mutations] [--debug] [--dry-run]

Error: Process completed with exit code 1.
```

--- a/.github/workflows/deptrac.yml
+++ b/.github/workflows/deptrac.yml
@@ -39,13 +39,12 @@ jobs:
           COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Get composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+        run: echo "COMPOSER_CACHE_FILES_DIR=$(composer config cache-files-dir)" >> $GITHUB_ENV
 
       - name: Cache composer dependencies
         uses: actions/cache@v3
         with:
-          path: ${{ steps.composer-cache.outputs.dir }}
+          path: ${{ env.COMPOSER_CACHE_FILES_DIR }}
           key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ hashFiles('**/composer.lock') }}
           restore-keys: ${{ runner.os }}-composer-
 

--- a/.github/workflows/infection.yml
+++ b/.github/workflows/infection.yml
@@ -45,13 +45,12 @@ jobs:
         uses: mheap/phpunit-matcher-action@v1
 
       - name: Get composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+        run: echo "COMPOSER_CACHE_FILES_DIR=$(composer config cache-files-dir)" >> $GITHUB_ENV
 
       - name: Cache composer dependencies
         uses: actions/cache@v3
         with:
-          path: ${{ steps.composer-cache.outputs.dir }}
+          path: ${{ env.COMPOSER_CACHE_FILES_DIR }}
           key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ hashFiles('**/composer.lock') }}
           restore-keys: ${{ runner.os }}-composer-
 
@@ -65,8 +64,6 @@ jobs:
 
       - name: Test with PHPUnit
         run: vendor/bin/phpunit --teamcity
-        env:
-          GITHUB_ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Mutate with Infection
         run: |

--- a/.github/workflows/infection.yml
+++ b/.github/workflows/infection.yml
@@ -64,6 +64,8 @@ jobs:
 
       - name: Test with PHPUnit
         run: vendor/bin/phpunit --teamcity
+        env:
+          GITHUB_ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Mutate with Infection
         run: |

--- a/.github/workflows/phpcsfixer.yml
+++ b/.github/workflows/phpcsfixer.yml
@@ -34,13 +34,12 @@ jobs:
           COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Get composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+        run: echo "COMPOSER_CACHE_FILES_DIR=$(composer config cache-files-dir)" >> $GITHUB_ENV
 
       - name: Cache composer dependencies
         uses: actions/cache@v3
         with:
-          path: ${{ steps.composer-cache.outputs.dir }}
+          path: ${{ env.COMPOSER_CACHE_FILES_DIR }}
           key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ hashFiles('**/composer.lock') }}
           restore-keys: ${{ runner.os }}-composer-
 

--- a/.github/workflows/phpstan.yml
+++ b/.github/workflows/phpstan.yml
@@ -43,13 +43,12 @@ jobs:
           COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Get composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+        run: echo "COMPOSER_CACHE_FILES_DIR=$(composer config cache-files-dir)" >> $GITHUB_ENV
 
       - name: Cache composer dependencies
         uses: actions/cache@v3
         with:
-          path: ${{ steps.composer-cache.outputs.dir }}
+          path: ${{ env.COMPOSER_CACHE_FILES_DIR }}
           key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ hashFiles('**/composer.lock') }}
           restore-keys: ${{ runner.os }}-composer-
 

--- a/.github/workflows/phpunit.yml
+++ b/.github/workflows/phpunit.yml
@@ -64,6 +64,7 @@ jobs:
         env:
           TERM: xterm-256color
           TACHYCARDIA_MONITOR_GA: enabled
+          GITHUB_ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - if: matrix.php-versions == '8.0'
         name: Run Coveralls

--- a/.github/workflows/phpunit.yml
+++ b/.github/workflows/phpunit.yml
@@ -42,13 +42,12 @@ jobs:
           COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Get composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+        run: echo "COMPOSER_CACHE_FILES_DIR=$(composer config cache-files-dir)" >> $GITHUB_ENV
 
       - name: Cache composer dependencies
         uses: actions/cache@v3
         with:
-          path: ${{ steps.composer-cache.outputs.dir }}
+          path: ${{ env.COMPOSER_CACHE_FILES_DIR }}
           key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ hashFiles('**/composer.lock') }}
           restore-keys: ${{ runner.os }}-composer-
 
@@ -65,7 +64,6 @@ jobs:
         env:
           TERM: xterm-256color
           TACHYCARDIA_MONITOR_GA: enabled
-          GITHUB_ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - if: matrix.php-versions == '8.0'
         name: Run Coveralls

--- a/.github/workflows/psalm.yml
+++ b/.github/workflows/psalm.yml
@@ -39,13 +39,12 @@ jobs:
           COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Get composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+        run: echo "COMPOSER_CACHE_FILES_DIR=$(composer config cache-files-dir)" >> $GITHUB_ENV
 
       - name: Cache composer dependencies
         uses: actions/cache@v3
         with:
-          path: ${{ steps.composer-cache.outputs.dir }}
+          path: ${{ env.COMPOSER_CACHE_FILES_DIR }}
           key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ hashFiles('**/composer.lock') }}
           restore-keys: ${{ runner.os }}-composer-
 

--- a/.github/workflows/rector.yml
+++ b/.github/workflows/rector.yml
@@ -43,13 +43,12 @@ jobs:
           COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Get composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+        run: echo "COMPOSER_CACHE_FILES_DIR=$(composer config cache-files-dir)" >> $GITHUB_ENV
 
       - name: Cache composer dependencies
         uses: actions/cache@v3
         with:
-          path: ${{ steps.composer-cache.outputs.dir }}
+          path: ${{ env.COMPOSER_CACHE_FILES_DIR }}
           key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ hashFiles('**/composer.lock') }}
           restore-keys: ${{ runner.os }}-composer-
 
@@ -63,5 +62,5 @@ jobs:
 
       - name: Analyze for refactoring
         run: |
-          composer global require --dev rector/rector:^0.14
+          composer global require --dev rector/rector:^0.15.1
           rector process --dry-run --no-progress-bar

--- a/.php-cs-fixer.dist.php
+++ b/.php-cs-fixer.dist.php
@@ -10,11 +10,19 @@ $finder = Finder::create()
         __DIR__ . '/app/',
         __DIR__ . '/tests/',
     ])
-    ->exclude('build')
-    ->exclude('Views/errors')
-    ->append([__FILE__]);
+    ->exclude([
+        'build',
+        'Views',
+    ])
+    ->append([
+        __FILE__,
+        __DIR__ . '/rector.php',
+    ]);
 
-$overrides = [];
+$overrides = [
+    // 'declare_strict_types' => true,
+    // 'void_return'          => true,
+];
 
 $options = [
     'finder'    => $finder,

--- a/app/Config/App.php
+++ b/app/Config/App.php
@@ -437,6 +437,7 @@ class App extends BaseConfig
      * Defaults to `Lax` as recommended in this link:
      *
      * @see https://portswigger.net/web-security/csrf/samesite-cookies
+     *
      * @deprecated `Config\Cookie` $samesite property is used.
      *
      * @var string

--- a/app/Libraries/Blog.php
+++ b/app/Libraries/Blog.php
@@ -250,7 +250,8 @@ class Blog
 
         foreach ($contents as $line) {
             if (trim($line) === '---') {
-                $inFrontMatter = ! $inFrontMatter;
+                $inFrontMatter = $inFrontMatter ? false : true;
+
                 if (! $inFrontMatter) {
                     $inBody = true;
                 }
@@ -267,6 +268,7 @@ class Blog
 
             $body[] = trim($line);
         }
+
         $post->body = implode("\n", $body);
 
         // Convert body using Markdown

--- a/app/Libraries/Blog.php
+++ b/app/Libraries/Blog.php
@@ -34,9 +34,9 @@ class Blog
      * If $category is present, will locate within a
      * subfolder of that name.
      *
-     * @throws BlogException
-     *
      * @return Post[]
+     *
+     * @throws BlogException
      */
     public function getRecentPosts(int $limit = 5, int $offset = 0, ?string $category = null)
     {

--- a/app/Libraries/GitHub.php
+++ b/app/Libraries/GitHub.php
@@ -100,7 +100,7 @@ class GitHub
         $this->client = $client;
     }
 
-    //---------------------------------------------------------------------
+    // ---------------------------------------------------------------------
 
     /**
      * Interfaces with the client to execute
@@ -127,7 +127,7 @@ class GitHub
         return $caller->{$final}(...$segments);
     }
 
-    //---------------------------------------------------------------------
+    // ---------------------------------------------------------------------
 
     /**
      * Retrieves releases and tags for configured repositories
@@ -244,7 +244,7 @@ class GitHub
         return $this->api(['repo', 'commits', 'show'], $segments);
     }
 
-    //---------------------------------------------------------------------
+    // ---------------------------------------------------------------------
 
     /**
      * Retrieves repo details for each configured repository as a Repo.
@@ -264,7 +264,7 @@ class GitHub
         return $this->storage['repos'];
     }
 
-    //---------------------------------------------------------------------
+    // ---------------------------------------------------------------------
 
     /**
      * Retrieves contributor information for a repository.

--- a/composer.json
+++ b/composer.json
@@ -66,10 +66,12 @@
             "@inspect",
             "@style"
         ],
+        "cs": "php-cs-fixer fix --ansi --verbose --dry-run --diff",
+        "cs-fix": "php-cs-fixer fix --ansi --verbose --diff",
+        "style": "@cs-fix",
         "deduplicate": "phpcpd app/ src/",
         "inspect": "deptrac analyze --cache-file=build/deptrac.cache",
         "mutate": "infection --threads=2 --skip-initial-tests --coverage=build/phpunit",
-        "style": "php-cs-fixer fix --verbose --ansi --using-cache=no",
         "test": "phpunit"
     }
 }

--- a/rector.php
+++ b/rector.php
@@ -1,6 +1,7 @@
 <?php
 
 use Rector\CodeQuality\Rector\BooleanAnd\SimplifyEmptyArrayCheckRector;
+use Rector\CodeQuality\Rector\Class_\CompleteDynamicPropertiesRector;
 use Rector\CodeQuality\Rector\Expression\InlineIfToExplicitIfRector;
 use Rector\CodeQuality\Rector\For_\ForToForeachRector;
 use Rector\CodeQuality\Rector\Foreach_\UnusedForeachValueToArrayKeysRector;
@@ -29,15 +30,23 @@ use Rector\Php55\Rector\String_\StringClassNameToClassConstantRector;
 use Rector\Php56\Rector\FunctionLike\AddDefaultValueForUndefinedVariableRector;
 use Rector\Php73\Rector\FuncCall\JsonThrowOnErrorRector;
 use Rector\Php73\Rector\FuncCall\StringifyStrNeedlesRector;
-use Rector\Php74\Rector\Property\TypedPropertyRector;
+use Rector\PHPUnit\Rector\Class_\AnnotationWithValueToAttributeRector;
 use Rector\PHPUnit\Set\PHPUnitSetList;
+use Rector\Privatization\Rector\Property\PrivatizeFinalClassPropertyRector;
 use Rector\PSR4\Rector\FileWithoutNamespace\NormalizeNamespaceByPSR4ComposerAutoloadRector;
 use Rector\Set\ValueObject\LevelSetList;
 use Rector\Set\ValueObject\SetList;
 
 return static function (RectorConfig $rectorConfig): void {
-    $rectorConfig->sets([SetList::DEAD_CODE, LevelSetList::UP_TO_PHP_74, PHPUnitSetList::PHPUNIT_SPECIFIC_METHOD, PHPUnitSetList::PHPUNIT_80]);
+    $rectorConfig->sets([
+        SetList::DEAD_CODE,
+        LevelSetList::UP_TO_PHP_74,
+        PHPUnitSetList::PHPUNIT_SPECIFIC_METHOD,
+        PHPUnitSetList::PHPUNIT_100,
+    ]);
+
     $rectorConfig->parallel();
+
     // The paths to refactor (can also be supplied with CLI arguments)
     $rectorConfig->paths([
         __DIR__ . '/app/',
@@ -73,13 +82,14 @@ return static function (RectorConfig $rectorConfig): void {
 
         // Note: requires php 8
         RemoveUnusedPromotedPropertyRector::class,
+        AnnotationWithValueToAttributeRector::class,
 
         // Ignore tests that might make calls without a result
         RemoveEmptyMethodCallRector::class => [
             __DIR__ . '/tests',
         ],
 
-        // Ignore files that should not be namespaced
+        // Ignore files that should not be namespaced to their folder
         NormalizeNamespaceByPSR4ComposerAutoloadRector::class => [
             __DIR__ . '/app/Helpers',
         ],
@@ -90,6 +100,10 @@ return static function (RectorConfig $rectorConfig): void {
         // May be uninitialized on purpose
         AddDefaultValueForUndefinedVariableRector::class,
     ]);
+
+    // auto import fully qualified class names
+    $rectorConfig->importNames();
+
     $rectorConfig->rule(SimplifyUselessVariableRector::class);
     $rectorConfig->rule(RemoveAlwaysElseRector::class);
     $rectorConfig->rule(CountArrayToEmptyArrayComparisonRector::class);
@@ -112,9 +126,7 @@ return static function (RectorConfig $rectorConfig): void {
     $rectorConfig->rule(MakeInheritedMethodVisibilitySameAsParentRector::class);
     $rectorConfig->rule(SimplifyEmptyArrayCheckRector::class);
     $rectorConfig->rule(NormalizeNamespaceByPSR4ComposerAutoloadRector::class);
-    $rectorConfig
-        ->ruleWithConfiguration(TypedPropertyRector::class, [
-            // Set to false if you use in libraries, or it does create breaking changes.
-            TypedPropertyRector::INLINE_PUBLIC => false,
-        ]);
+    $rectorConfig->rule(StringClassNameToClassConstantRector::class);
+    $rectorConfig->rule(PrivatizeFinalClassPropertyRector::class);
+    $rectorConfig->rule(CompleteDynamicPropertiesRector::class);
 };

--- a/rector.php
+++ b/rector.php
@@ -99,6 +99,11 @@ return static function (RectorConfig $rectorConfig): void {
 
         // May be uninitialized on purpose
         AddDefaultValueForUndefinedVariableRector::class,
+
+        // PHPStan will report false positive errors
+        UnnecessaryTernaryExpressionRector::class => [
+            __DIR__ . '/app/Libraries/Blog.php',
+        ],
     ]);
 
     // auto import fully qualified class names

--- a/tests/entities/ContributorTest.php
+++ b/tests/entities/ContributorTest.php
@@ -18,7 +18,7 @@ final class ContributorTest extends ProjectTestCase
         $this->assertSame($expected, $contributor->stars);
     }
 
-    public function contributionsProvider(): array
+    public static function contributionsProvider(): array
     {
         return [
             [0, '★'],

--- a/tests/entities/PostTest.php
+++ b/tests/entities/PostTest.php
@@ -8,7 +8,7 @@ use Tests\Support\ProjectTestCase;
  */
 final class PostTest extends ProjectTestCase
 {
-    protected Post $post;
+    private Post $post;
 
     protected function setUp(): void
     {

--- a/tests/libaries/BlogLibraryTest.php
+++ b/tests/libaries/BlogLibraryTest.php
@@ -4,7 +4,6 @@ use App\Exceptions\BlogException;
 use App\Libraries\Blog;
 use CodeIgniter\Config\Factories;
 use Config\Blog as BlogConfig;
-use org\bovigo\vfs\vfsStream;
 use Tests\Support\ProjectTestCase;
 
 /**
@@ -12,12 +11,7 @@ use Tests\Support\ProjectTestCase;
  */
 final class BlogLibraryTest extends ProjectTestCase
 {
-    /**
-     * Our fake filesystem
-     */
-    protected vfsStream $root;
-
-    protected Blog $blog;
+    private Blog $blog;
 
     protected function setUp(): void
     {

